sharding/indexHashedNodesCoordinator.go

```diff
@@ -84,17 +95,24 @@ func (ihgs *indexHashedNodesCoordinator) SetNodesPerShards(nodes map[uint32][]Va
 //    the item at the new proposed index is not found in the list. This new proposed index will be called checked index
 // 4. the item at the checked index is appended in the temp validator list
 func (ihgs *indexHashedNodesCoordinator) ComputeValidatorsGroup(randomness []byte) (validatorsGroup []Validator, err error) {
-	if len(ihgs.nodesMap[ihgs.shardId]) < ihgs.consensusGroupSize {
-		return nil, ErrSmallEligibleListSize
-	}
-
 	if randomness == nil {
 		return nil, ErrNilRandomness
 	}
 
 	tempList := make([]Validator, 0)
+	var consensusGroupSize int
 
-	for startIdx := 0; startIdx < ihgs.consensusGroupSize; startIdx++ {
+	if ihgs == nil {
+		return nil, ErrNilRandomness
+	}
+
+	if ihgs.shardId == MetachainShardId {
```

Comment:
make a local function out of this if, as it is used in multiple places.

Reply:
done